### PR TITLE
[TOPI] Batch Norm Training Mode

### PR DESCRIPTION
**Pull request by SiriusNEO**
Prior to this PR, TOPI batch_norm only supports inference. 

This PR adds `training: bool` flag and `momentum: float` argument to support training mode (update `moving_mean / var` and return), which aligns with `torch.nn.functional.batch_norm`.

### CODE REVIEW

**Comment by SiriusNEO** [comment]
cc @MasterJH5574 @tqchen

**Review by MasterJH5574 — APPROVED** [review]
LGTM. Thanks for the enhancement!
